src/main/java/reposense/commits/CommitInfoAnalyzer.java

```diff
@@ -93,9 +104,61 @@ public static CommitResult analyzeCommit(CommitInfo commitInfo, RepoConfiguratio
-        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, insertion, deletion);
+        if (statLine.isEmpty()) { // empty commit, no files changed
+            return new CommitResult(author, hash, date, messageTitle, messageBody, tags, 0, 0, new HashMap<>());
+        }
+
+        String[] statInfos = statLine.split(NEW_LINE_SPLITTER);
+        String[] fileTypeContributions = Arrays.copyOfRange(statInfos, 0, statInfos.length - 1);
+        Map<FileType, ContributionPair> fileTypeAndContributionMap =
+                getFileTypesAndContribution(fileTypeContributions, config);
+
+        String contributionStat = statInfos[statInfos.length - 1]; // last index is the file contribution statistics
+        int totalCommitInsertions = getInsertion(contributionStat);
+        int totalCommitDeletions = getDeletion(contributionStat);
+
+        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, totalCommitInsertions,
+                totalCommitDeletions, fileTypeAndContributionMap);
+    }
+
+    /**
+     * Extract the additions and deletions of file types that have been modified
+     */
+    private static Map<FileType, ContributionPair> getFileTypesAndContribution(String[] filePathContributions,
+            RepoConfiguration config) {
+        Map<FileType, ContributionPair> fileTypesAndContributionMap = new HashMap<>();
+        for (String filePathContribution : filePathContributions) {
+            String[] infos = filePathContribution.split(TAB_SPLITTER);
+            int addition = Integer.parseInt(infos[STAT_ADDITION_INDEX]);
+            int deletion = Integer.parseInt(infos[STAT_DELETION_INDEX]);
+            String filePath = extractFilePath(infos[STAT_FILE_PATH_INDEX]);
+            FileType fileType = config.getFileType(filePath);
+
+            if (!fileTypesAndContributionMap.containsKey(fileType)) {
+                fileTypesAndContributionMap.put(fileType, new ContributionPair());
+            }
+
+            ContributionPair contributionPair = fileTypesAndContributionMap.get(fileType);
+            contributionPair.addInsertions(addition);
+            contributionPair.addDeletions(deletion);
+        }
+        return fileTypesAndContributionMap;
+    }
+
+    /**
+     * Extracts the correct file path from the unprocessed git log {@code filePath}
+     */
+    private static String extractFilePath(String filePath) {
+        if (!filePath.contains(MOVED_FILE_INDICATION)) {
+            return filePath;
+        }
+        // moved file has the format: fileA => newPosition/fileA
+        String filteredFilePath =
+                filePath.substring(filePath.indexOf(MOVED_FILE_INDICATION) + MOVED_FILE_INDICATION.length());
+        if (filteredFilePath.charAt(filteredFilePath.length() - 1) == '}') { // renamed file has ending '}' char
+            filteredFilePath = filteredFilePath.substring(0, filteredFilePath.length() - 1);
+        }
```

Comment:
```suggestion
        // Removes the trailing '}' character from the file name if it exists.
        filteredFilePath = filteredFilePath.replaceAll("}$", "");
```

Btw, will there be any issue when a file is renamed to one ending with a `}`?

Reply:
Yes, that may happen.

But the filePath is only used for getting the filetype, so its fine to remove the ending `}`, as a file type that includes `}` at the end wouldn't make sense.